user_interface/src/store/modules/home.js

```diff
@@ -55,8 +56,17 @@ const mutations = {
   setSelectMode: (state, selectMode) => {
     state.selectMode = selectMode;
   },
-  setSelectNumber: (state, selectNumber) => {
-    state.selectNumber = selectNumber;
+  addBoxIdToSelected: (state, boxId) => {
+    state.selectNumberToBoxIdArray.push(boxId);
```

Comment:
I read before that for states, it may be better to reassign instead of manipulating directly.
Can you help me do that for Map() too? Tq :>

<pre>
state.selectNumberToBoxIdArray = [...state.selectNumberToBoxIdArray, boxId];
</pre>

Reply:
I changed to this for the array, but the spread syntax doesn't work for Map(), it converts the Map to an array. I don't think it should be a big issue since I'm not actually doing any reactive updates based on the Map() mutations, and also the Vue docs say that mutations are synchronous, so there shouldn't be any race conditions where the hold's text labels are update before the Map() is updated.